packages/@aws-cdk/aws-cloudfront/test/distribution.test.ts

```diff
@@ -475,61 +471,25 @@ describe('certificates', () => {
     }).toThrow(/Must specify at least one domain name/);
   });
 
-  describe('adding a certificate and domain renders the correct ViewerCertificate and Aliases property', () => {
-    testFutureBehavior(
-      'when @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021 is enabled, use the TLSv1.2_2021 security policy by default',
-      { [CLOUDFRONT_DEFAULT_SECURITY_POLICY_TLS_V1_2_2021]: true },
-      App,
-      (customApp) => {
-        const customStack = new Stack(customApp);
-
-        const certificate = acm.Certificate.fromCertificateArn(customStack, 'Cert', 'arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012');
-
-        new Distribution(customStack, 'Dist', {
-          defaultBehavior: { origin: defaultOrigin() },
-          domainNames: ['example.com', 'www.example.com'],
-          certificate,
-        });
-
-        Template.fromStack(customStack).hasResourceProperties('AWS::CloudFront::Distribution', {
-          DistributionConfig: {
-            Aliases: ['example.com', 'www.example.com'],
-            ViewerCertificate: {
-              AcmCertificateArn: 'arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012',
-              SslSupportMethod: 'sni-only',
-              MinimumProtocolVersion: 'TLSv1.2_2021',
-            },
-          },
-        });
-      },
-    );
-
-    testLegacyBehavior(
-      'when @aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021 is disabled, use the TLSv1.2_2019 security policy by default',
-      App,
-      (customApp) => {
-        const customStack = new Stack(customApp);
-
-        const certificate = acm.Certificate.fromCertificateArn(customStack, 'Cert', 'arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012');
-
-        new Distribution(customStack, 'Dist', {
-          defaultBehavior: { origin: defaultOrigin() },
-          domainNames: ['example.com', 'www.example.com'],
-          certificate,
-        });
-
-        Template.fromStack(customStack).hasResourceProperties('AWS::CloudFront::Distribution', {
-          DistributionConfig: {
-            Aliases: ['example.com', 'www.example.com'],
-            ViewerCertificate: {
-              AcmCertificateArn: 'arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012',
-              SslSupportMethod: 'sni-only',
-              MinimumProtocolVersion: 'TLSv1.2_2019',
-            },
-          },
-        });
+  test('use the TLSv1.2_2021 security policy by default', () => {
```

Comment:
Don't we still need to test the legacy behavior though? For some of the feature flags it is possible to still set the flag to `false`.

Not sure if it would be better to keep `testLegacyBehavior` or just set the feature flag through `node.setContext()`

Reply:
There are places that the feature flags are set as false to do that very thing. testLegacyBehavior, however, just skips the tests in v2, it doesn't set the flags to false.